controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go

```diff
@@ -57,6 +59,15 @@ func (r *ClusterReconciler) ReconcileAddonSecret(
 		}
 	}
 
+	//Stop reconciling if AKO ip family doesn't match cluster ip family
+	if err = validateADCAndClusterIpFamily(cluster, obj, isVIPProvider, log); err != nil {
+		errInfo := "Selected AKODeploymentConfig " + obj.Name + "'s IP family doesn't match cluster " + cluster.Namespace +
+			"-" + cluster.Name + "'s ip family, stop deploying AKO into cluster " + cluster.Namespace + "-" + cluster.Name
+		log.Error(err, errInfo)
+		conditions.MarkTrue(cluster, akoov1alpha1.AKOIpFamilyValidationFailedCondition)
```

Comment:
I believe this should be `MarkFalse` ?

Reply:
The condition is AKOIpFamilyValidationFailed. I think when it marks `true`, it means the validation fails.

Reply:
That will be super confusing, I think it is better to change the condition name to something like:
```
AKOIpFamilyValidation
```

Then you can mark that one false, btw, can you add a message in this condition? A detailed description